### ISSUE TITLE
A new user with temporary password can not log in successfully after changing password.

### ISSUE THREAD

**Issue by ikeough** [hypar-io/sdk#35]
I've attempted the following to reproduce:
- Create a new user, specifying the user name, email, and a temporary password. Unselected "Mark email as verified" to force the user creating a new password through the CLI.
- `hypar publish` to get a login prompt.
  - Insert user name.
  - Insert temporary password. (I had to do this a couple of times because cut and paste of password was inserting some garbage characters from my scratch pad.)
  - Prompted to enter new password.
  - Entered new password.
  - Login completes.

Thinking that perhaps there was some AWS configuration stuff that was still being used by the login process, I removed my aws credentials file altogether, and retried the steps above with another test user. And it succeeded.
![image](https://user-images.githubusercontent.com/1139788/42969490-31f4ba76-8b5b-11e8-87a0-ee43d599aa26.png)

Verified that this fails the same way using the Linux and Windows CLIs.

**Comment by ikeough** [comment]
After attempting to login, the user receives:
```
Login to Hypar failed.
The publish command was not recognized. Try 'hypar help'.
```

This is a red herring. The failed login attempt is making `PublishCommand.CanExecute` return false. This does require a fix though.

**Comment by ikeough** [comment]
![image](https://user-images.githubusercontent.com/1139788/42970065-1fb8867e-8b5d-11e8-8e3f-1a63ce193b46.png)

**Comment by ikeough** [comment]
Cognito has policies in place for password requirements:
![image](https://user-images.githubusercontent.com/1139788/42970233-9eb0c9dc-8b5d-11e8-9687-395681d674f6.png)
It's possible that the new password supplied by the user doesn't meet these criteria and we're not handling the response.

**Comment by ikeough** [comment]
This is partially fixed in ... Now when a user attempts to enter a new password that doesn't conform to the policy, they receive the following:

```
Enter your user name:
test3
Enter your password:
************
	waiting...
	Enter your desired new password:
*****
Passwords must be a minimum of 8 characters, and contain upper and lower case letters, a number, and a special character.
The publish command was not recognized. Try 'hypar help'.
```

**Comment by ikeough** [comment]
This is fixed in 4c044c86f571a4266996231092a525fc0c89cdc9.
